src/bus/lattice.rs

```diff
@@ -77,6 +96,92 @@ impl DistributedBus {
     }
 }
 
+fn spawn_inventory_handler(
+    nc: Arc<nats::Connection>,
+    host_id: String,
+    claims: Arc<RwLock<HashMap<String, Claims<Actor>>>>,
+    bindings: Arc<RwLock<BindingsList>>,
+    caps: Arc<RwLock<HashMap<RouteKey, CapabilityDescriptor>>>,
+) -> Result<()> {
+    let _ = nc.subscribe(INVENTORY_SUBJECT)?.with_handler(move |msg| {
+        trace!("Handling Inventory Request");
+        match msg.subject.clone().as_str() {
+            INVENTORY_HOSTS => respond_with_host(msg, host_id.to_string()),
+            INVENTORY_ACTORS => respond_with_actors(msg, host_id.to_string(), claims.clone()),
+            INVENTORY_BINDINGS => respond_with_bindings(msg, host_id.to_string(), bindings.clone()),
+            INVENTORY_CAPABILITIES => respond_with_caps(msg, host_id.to_string(), caps.clone()),
+            _ => Err(std::io::Error::new(
+                std::io::ErrorKind::Other,
+                "Bad inventory topic!",
+            )),
+        }
+    });
+    Ok(())
+}
+
+fn respond_with_host(
+    msg: nats::Message,
+    host_id: String,
+) -> std::result::Result<(), std::io::Error> {
+    msg.respond(serde_json::to_vec(&InventoryResponse::Host(host_id)).unwrap())
+}
+
+fn respond_with_actors(
+    msg: nats::Message,
+    host: String,
+    claims: Arc<RwLock<HashMap<String, Claims<Actor>>>>,
+) -> std::result::Result<(), std::io::Error> {
+    let ir = InventoryResponse::Actors {
+        host,
+        actors: claims.read().unwrap().values().cloned().collect(),
+    };
+    msg.respond(serde_json::to_vec(&ir).unwrap())
+        .map_err(|e| e.into())
+}
+
+fn respond_with_bindings(
+    msg: nats::Message,
+    host: String,
+    bindings: Arc<RwLock<BindingsList>>,
+) -> std::result::Result<(), std::io::Error> {
+    let mut items = Vec::<Binding>::new();
+    let lock = bindings.read().unwrap();
+    for (k, v) in lock.iter() {
+        items.push(Binding {
+            actor: k.0.to_string(),
+            capability_id: k.1.to_string(),
+            binding_name: k.2.to_string(),
+            configuration: v.values.clone(),
+        });
+    }
+    let ir = InventoryResponse::Bindings {
+        host,
+        bindings: items,
+    };
+    msg.respond(serde_json::to_vec(&ir).unwrap())
+        .map_err(|e| e.into())
+}
+
+fn respond_with_caps(
+    msg: nats::Message,
+    host: String,
+    caps: Arc<RwLock<HashMap<RouteKey, CapabilityDescriptor>>>,
+) -> std::result::Result<(), std::io::Error> {
+    let mut capabilities = vec![];
```

Comment:
Here you use `vec![]`, above (line 147) you use Vec::<Binding>::new(), I know this is nitpicky but do we have a preferred method to instantiate a new empty vector?

Reply:
I usually prefer `Vec::new()`. From the compiler's perspective, `vec![]` and `Vec::new()` are identical (the macro expands to the latter).

I think what we should ultimately be looking for is opportunities to use `Vec::with_capacity`. If we know the number of items we're going to have, or even a ballpark number, then we might be able to save some _time_ by pre-allocating in bulk.